meta-edgeos/recipes-support/expand-rootfs/files/expand-rootfs.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+STAMP="/var/lib/expand-rootfs.done"
+LOG="/var/log/expand-rootfs.log"
+
+# Redirect all output to log file and console
+exec > >(tee -a "$LOG") 2>&1 || true
+echo "[expand-rootfs] Start $(date -Is || true)"
+
+# Run only once
+if [[ -f "$STAMP" ]]; then
+  echo "[expand-rootfs] Stamp file exists; exiting."
+  exit 0
+fi
+
+# Detect root device and filesystem type
+ROOTDEV="$(findmnt -n -o SOURCE /)"
+FSTYPE="$(findmnt -n -o FSTYPE /)"
+
+if [[ "$ROOTDEV" != /dev/* ]]; then
+  echo "[expand-rootfs] Root device is not a physical partition ($ROOTDEV) -> skip."
+  mkdir -p "$(dirname "$STAMP")"; touch "$STAMP"; exit 0
+fi
+
+# Use sysfs to find partition number and parent disk (portable)
+PART_BASENAME="$(basename "$ROOTDEV")"   # e.g. mmcblk0p2
+PARTNUM="$(cat "/sys/class/block/${PART_BASENAME}/partition")"
+PARENT_SYS="$(readlink -f "/sys/class/block/${PART_BASENAME}/..")"
+DISK="/dev/$(basename "$PARENT_SYS")"    # e.g. /dev/mmcblk0
+echo "[expand-rootfs] RootDEV=$ROOTDEV Disk=$DISK Part=$PARTNUM FS=$FSTYPE"
+
+# Skip if an A/B rootfs layout is detected (e.g. mender)
+if lsblk -rno PARTLABEL "$DISK" 2>/dev/null | grep -qiE 'root[a-b]'; then
```

Comment:
this part will be interesting. we do want to get to an A/B setup, but at that point I think we'll have a separate partition for data so updates to the os don't wipe everything out.